core/signer/signer.go

```diff
@@ -115,20 +125,42 @@ func (s *Signer) ProcessMutation(ctx context.Context, txn transaction.Txn, index
 		return fmt.Errorf("QueueLeaf err: %v", err)
 	}
 
-	mutationObj := new(tpb.SignedKV)
-	if err := proto.Unmarshal(mutation, mutationObj); err != nil {
-		return err
-	}
-	if _, err := s.mutations.Write(txn, mutationObj); err != nil {
-		return fmt.Errorf("Mutation write failed: %v", err)
-	}
-
 	log.Printf("Sequenced %x", index)
 	return nil
 }
 
+// processMutations reads mutations from the database and adds them to the tree.
+// processMutations returns the maximum mutation sequence number processed.
+func (s *Signer) processMutations(ctx context.Context, txn transaction.Txn) (uint64, error) {
+	startSequence := uint64(0)
+	smh := new(ctmap.SignedMapHead)
+	if _, _, err := s.sths.Latest(ctx, smh); err == nil {
+		startSequence = smh.GetMapHead().MaxSequenceNumber + 1
+	} else if err != sql.ErrNoRows {
+		return 0, fmt.Errorf("Latest err: %v", err)
+	}
+	maxSequence, mutations, err := s.mutations.ReadRange(txn, startSequence, math.MaxInt32)
+	if err != nil {
+		return 0, fmt.Errorf("ReadRange err: %v", err)
+	}
+	for _, mutation := range mutations {
+		mData, err := proto.Marshal(mutation)
```

Comment:
Mutation APIs take and return `SignedKV` and not `[]byte`, do we want to use `interface{}` instead?